addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/handler/AirVisualNodeHandler.java

```diff
@@ -0,0 +1,267 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (Exception e) {
+            logger.debug("Unexpected error connecting to Node", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void pollNode() throws IOException {
+        String jsonData = getNodeJsonData();
+        NodeData currentNodeData = gson.fromJson(jsonData, NodeData.class);
+        if (nodeData == null || currentNodeData.getStatus().getDatetime() > nodeData.getStatus().getDatetime()) {
+            nodeData = currentNodeData;
+            // Update all channels from the updated Node data
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId(), false);
+            }
+        }
+    }
+
+    private String getNodeJsonData() throws IOException {
+        String url = "smb://" + nodeAddress +"/" + nodeShareName + "/" + NODE_JSON_FILE;
+        NtlmPasswordAuthentication auth = new NtlmPasswordAuthentication(null, nodeUsername, nodePassword);
+        try (SmbFileInputStream in = new SmbFileInputStream(new SmbFile(url, auth))) {
+            return IOUtils.toString(in, StandardCharsets.UTF_8.name());
+        }
+    }
+
+    private void updateChannel(String channelId, boolean force) {
+        if (force || isLinked(channelId)) {
+            Object value = getChannelValue(channelId, nodeData);
+
+            // Get channel value
+            State state = null;
+            if (value == null) {
+                state = UnDefType.UNDEF;
+            } else if (value instanceof Calendar) {
+                state = new DateTimeType((Calendar) value);
+            } else if (value instanceof Integer) {
+                state = new DecimalType(BigDecimal.valueOf(((Integer) value).longValue()));
+            } else if (value instanceof Float) {
+                state = new DecimalType(BigDecimal.valueOf(((Float) value).floatValue()).setScale(1,
+                        RoundingMode.HALF_UP));
+            } else if (value instanceof String) {
+                state = new StringType(value.toString());
+            } else {
+                logger.warn("Update channel {}: Unsupported value type {}", channelId,
+                        value.getClass().getSimpleName());
+            }
+
+            // Update the channel
+            if (state != null) {
+                logger.debug("Update channel {} with state {} ({})", channelId, state.toString(),
+                        (value == null) ? "null" : value.getClass().getSimpleName());
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    private Object getChannelValue(String fullChannelId, NodeData nodeData) {
+        Object value = null;
+
+        if (nodeData != null) {
+            String[] fullChannelIdFields = StringUtils.split(fullChannelId, "#");
```

Comment:
I do not understand why you have to split the channelID by `#`. You should pass the correct ID to this method instead, splitting at `#` is not reliable.